gateway/assets/script/bootstrap.js

```diff
@@ -603,4 +600,8 @@ function uuidv4() {
     return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, function(c) {
         return (c ^ cryptoInstance.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
     })
-}  
\ No newline at end of file
+}
+
+function getNamespaceSuffix(namespace) {
```

Comment:
this is a small complaint, but it looks like this returns a query parameter, not a suffix.  It also requires this method to be called first because it returns the `?`, this will make it a little fragile is someone doesn't recognize that the order is important. Javascript has a URL object https://developer.mozilla.org/en-US/docs/Web/API/URL that might be easier to use and more robust to work with, since the search params object has a set method https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/set

Reply:
@LucasRoesler 
The URL object doesn't support relative paths, and the CORS blocking on localhost means that using `new URL(document.lcoation.domain)` to generate a url `localhost:8080` and then using the url object to do this stuff doesnt work because (at least) chrome block the requests (when i tried anyway) a rename and re-structure of this is incoming. Hopefully its better.

Reply:
wow that is an incredibly disappointing API in JS then :(

Reply:
I mean it worked, but because mostly people will be using localhost CORS gets in the way and makes things hard